packages/scripts/src/upgrade-tests/Proposer.js

```diff
@@ -0,0 +1,137 @@
+      ZERO_ADDRESS
+    ).send({ from: defaultAccount });
+  }
+
+  // Grab the current owner and proposer accounts from the governor.
+  const governorOwner = await governor.methods.getMember(0).call();
+  const governorProposer = await governor.methods.getMember(1).call();
+
+  // As with other accounts, the owner account will need to be given ETH and impersonated.
+  await addEth(governorOwner);
+  await impersonateAccount(governorOwner);
+
+  // Random large UMA holder (EOA). This account will be used as a source of UMA.
+  const largeHolder = "0x8bd16DE8938A2ad16794ac7E7502896f156E370C";
+  await impersonateAccount(largeHolder);
+
+  // Do upgrade to decentralized proposal system.
+  await governor.methods.resetMember(1, proposer.options.address).send({ from: governorOwner });
+  await governor.methods.resetMember(0, governor.options.address).send({ from: governorOwner });
+
+  // Send 10k tokens to the default account so it can make two proposals.
+  await votingToken.methods.transfer(defaultAccount, web3.utils.toWei("10000")).send({ from: largeHolder });
+
+  // Approve the proposal contract to take 10k tokens from the default wallet.
+  await votingToken.methods.approve(proposer.options.address, web3.utils.toWei("10000")).send({ from: defaultAccount });
+
+  // Create arbitrary transaction to test that the proposal system works for vanilla requests.
+  const usdc = await ERC20.at("0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48");
+  const approveTxnData = usdc.methods.approve(defaultAccount, 100).encodeABI();
+
+  // Propose the sample transaction and simulate a vote.
+  await proposer.methods
+    .propose([{ to: usdc.options.address, value: 0, data: approveTxnData }])
+    .send({ from: defaultAccount });
+  await simulateVote();
+
+  // Verify that the transaction went through.
+  assert(
+    (await usdc.methods.allowance(governor.options.address, defaultAccount).call()).toString() === "100",
+    "❌ Initial test transaction didn't go through!"
+  );
+
+  console.log("✅ Test proposal went through using new proposal system.");
+
+  // Make a proposal to reset the governor permissions to how they started.
+  const resetPermissionsProposal = [
+    { to: governor.options.address, data: governor.methods.resetMember(1, governorProposer).encodeABI(), value: 0 },
+    { to: governor.options.address, data: governor.methods.resetMember(0, governorOwner).encodeABI(), value: 0 },
+  ];
+
+  await proposer.methods.propose(resetPermissionsProposal).send({ from: defaultAccount });
+  await simulateVote();
+
+  assert(
+    (await governor.methods.getMember(0).call()) === governorOwner,
+    "❌ Owner wasn't set reset. Revert proposal wasn't successful."
+  );
+  assert(
+    (await governor.methods.getMember(1).call()) === governorProposer,
+    "❌ Proposer not reset. Revert proposal wasn't successful."
+  );
+
+  console.log("✅ Proposal to revert upgrade to decentralized proposal system succeded.");
+}
+
+async function impersonateAccount(account) {
```

Comment:
this is a useful method! might be nice to have it set as a common method but it is a bit confusing though as this only works in the hre context. would be nice to append it somehow to the hre object so it's always nicely useable...not something to worry about now

Reply:
Yeah, I agree. We could probably write a hardhat plugin to tack this onto the hre object.